conda/reporters.py

```diff
@@ -19,19 +22,29 @@
 if TYPE_CHECKING:
     from typing import Callable
 
+    from .plugins.types import ProgressBarBase
+
 logger = logging.getLogger(__name__)
 
 
-@lru_cache(maxsize=None)
-def _get_render_func(style: str | None = None) -> Callable:
+def _get_reporter_backend() -> str:
     """
-    Retrieves the render function to use
+    Determine the current reporter backend being used
     """
     if context.json:
         backend = DEFAULT_JSON_REPORTER_BACKEND
     else:
         backend = context.console
 
+    return backend
```

Comment:
This feels like something `context` should do for us, there's already several examples of similar things being done (e.g., `context.*_threads`):

```python
class Context(Configuration):
    _console = ParameterLoader(
        PrimitiveParameter(DEFAULT_CONSOLE_REPORTER_BACKEND, element_type=str),
        aliases=["console"],
    )

    @property
    def console(self) -> str:
        if self.json:
            return DEFAULT_JSON_REPORTER_BACKEND
        return self._console
```